app-frontend/src/app/components/tools/toolCreateModal/toolCreateModal.controller.js

```diff
@@ -126,6 +126,84 @@ export default class ToolCreateModalController {
         return builtNode;
     }
 
+    expressionTreeToMAML(tree, value, allowCollapse = true) {
```

Comment:
This could use some commenting to provide guideposts to a later reader; the code looks fine as-is, but without a deep knowledge of both MAML and mathjs it's tough to understand why it does some of the things that it does.